timelineGenerator.js

```diff
@@ -0,0 +1,233 @@
+import fs from 'fs';
+import path from 'path';
+import * as d3 from 'd3';
+import { JSDOM } from 'jsdom';
+
+const outputPath = 'gemRepository/integratedModelsTimeline.svg';
+const d3Curve = d3.line().curve(d3.curveNatural);
+const WIDTH = 1500;
+let GEMS, HEIGHT, GEM_POSITIONS, BRANCH_POINTS, TIME_SCALE;
+
+const createTimelineChart = async (integratedModelsPath) => {
+  GEMS = setGems(integratedModelsPath);
+  HEIGHT = getHeight();
+  GEM_POSITIONS = getGemPositions();
+  BRANCH_POINTS = getBranchPoints();
+  TIME_SCALE = createTimeScale();
+
+  const fakeDom = new JSDOM('<!DOCTYPE html><html><body></body></html>');
+  const body = (await d3).select(fakeDom.window.document).select('body');
+
+  // + 1 is needed or else the bottom axis appears hidden
+  let svg = body
+    .append('svg')
+    .attr('width', WIDTH)
+    .attr('height', HEIGHT + 1);
+
+  svg = addBranchPoints(svg);
+  svg = addGemVersions(svg);
+  svg = addTimeAxis(svg);
+  svg = addGemsAxis(svg);
+
+  fs.writeFileSync(outputPath, body.html());
+};
+
+const setGems = (integratedModelsPath) => {
+  const gems = [];
+
+  const intputDirFiles = fs.readdirSync(integratedModelsPath);
+  for (let i = 0; i < intputDirFiles.length; i++) {
+    const filePath = path.join(integratedModelsPath, intputDirFiles[i]);
+    const stat = fs.lstatSync(filePath);
+    if (stat.isDirectory() && intputDirFiles[i][0] != '.') {
+      const rawJson = fs.readFileSync(`${filePath}/gemRepository.json`, 'utf8');
+      gems.push(JSON.parse(rawJson));
+    }
+  }
+
+  return gems.map((versions) =>
+    versions.filter((v) => v.id.split('.')[2] === '0'),
+  );
+};
+
+const getHeight = () => GEMS.length * 50 + 100;
+
+const getModelFromId = (id) => id.split('-')[0];
+
+const getGemPositions = () =>
+  GEMS.reduce(
+    (acc, [v], i) => ({
```

Comment:
Minor thing but not sure at glance what `acc` refers to

Reply:
This is commonly used in reduce functions and it is short for accumulator ([example](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce#grouping_objects_by_a_property)). I do think it's better to use something more descriptive though and have added a commit here: https://github.com/MetabolicAtlas/data-generation/pull/27/commits/1b16575ce36b3471c242a0e42af1e4d94699e393